### PR TITLE
feat(graphql): compose queries from multiple sources

### PR DESCRIPTION
**Pull request by amnn**
## Description

GraphQL operations often share fragments, but `graphql_query!` previously accepted one literal, forcing operations and reusable fragments into the same source.

Allow comma-separated inline literals and `@"..."` file sources. Concatenate all sources before parsing, validation, and canonical formatting. File paths follow `include_str!` semantics by resolving relative to the invoking Rust source file, and generated `include_str!` sentinels keep files in Cargo's dependency graph.

## Test Plan

Added compile-pass coverage for multiple literals, including a token split across source boundaries. Added integration coverage for file-only and mixed inline/file queries.

### CODE REVIEW

File: crates/sui-graphql-macros/src/query.rs [hunk 1]
```diff
@@ -154,3 +136,100 @@ where
...
+        Some(SCHEMA_DIAGNOSTIC_LABEL),
+        errors,
+    );
+
+    format!("Failed to {action} Sui GraphQL schema:\n{formatted}")
+}
+
+/// Interpret `path` as a path relative to the source file containing the macro invocation.
+fn source_relative(literal: &LitStr) -> Result<PathBuf, syn::Error> {
+    let value = literal.value();
```
Comment (tpham-mysten) [inline crates/sui-graphql-macros/src/query.rs:220]: nit: maybe just inline this?

File: crates/sui-graphql-macros/src/query.rs [hunk 2]
```diff
@@ -154,3 +136,100 @@ where
 
     combined
 }
+
+fn expand_impl(input: TokenStream) -> Result<TokenStream2, syn::Error> {
+    let sources = Punctuated::<Source, Token![,]>::parse_terminated.parse(input)?;
+    if sources.is_empty() {
+        return Err(syn::Error::new(
+            proc_macro2::Span::call_site(),
+            "expected at least one GraphQL source",
```
Comment (tpham-mysten) [inline crates/sui-graphql-macros/src/query.rs:145]: nit: add test for empty case

File: crates/sui-graphql-macros/src/query.rs [hunk 3]
```diff
@@ -154,3 +136,100 @@ where
...
+        ));
+    }
+
+    // Concatenate exactly like Rust's `concat!`: callers can split a document
+    // wherever they like, and fragments can live in separate sources. The
+    // complete document is still parsed, validated, and formatted as one unit.
+    let mut source = String::new();
+    let mut dependencies = Vec::new();
+    for input in sources {
```
Comment (tpham-mysten) [inline crates/sui-graphql-macros/src/query.rs:154]: optional tiny nit: 

Maybe `for source in sources`?

File: crates/sui-graphql-macros/src/query.rs [hunk 4]
```diff
@@ -154,3 +136,100 @@ where
...
+                    syn::Error::new(
+                        literal.span(),
+                        format!(
+                            "failed to read GraphQL source from '{}': {error}",
+                            path.display()
+                        ),
+                    )
+                })?;
+
+                source.push_str(&contents);
```
Comment (tpham-mysten) [inline crates/sui-graphql-macros/src/query.rs:169]: If the file does not contain newline at EOF, it might be a problem. For example, if there is a comment at the end of a file like:

```
fragment ChainIdentifier on Query {
    chainIdentifier
}
# comment 
```
. When we concatenate with `query { ... }`, it will become

`..# commentquery { ... }`

**Review by tpham-mysten — APPROVED** [review]
Sorry for the delay, but it looks great! just a few nits

**Review by bmwill — APPROVED** [review]

**Comment by amnn** [comment]
Thanks @tpham-mysten, I've addressed these comments in #313
